### PR TITLE
Fix for JCLOUDS-655

### PR DESCRIPTION
**Pull request by monitorjbl**
First time committing to jclouds, let me know if there are any style guidelines I missed with this change.

### CODE REVIEW

**Comment by monitorjbl** [comment]
Haven't heard anything for a while, any updates?

**Comment by everett-toews** [comment]
Can you add that test code as a commit so we can see it? (don't squash it)

**Comment by monitorjbl** [comment]
Done. It throws an error saying "links" is null, which is to be expected since it's not defined the way that a normal Gson adapter would read the `Image` object from JSON. The trouble I'm having is testing this is an isolated enough way that it's still a unit test of just the adapter; it doesn't appear to work with any sample output from Openstack on its own.

**Comment by everett-toews** [comment]
@monitorjbl Sorry I haven't been able to have a look at this. I've been fighting the flu. 

@nacx Can you help with the unit test of just the adapter? If not, I can look at it later this week or next.

**Comment by nacx** [comment]
@monitorjbl @everett-toews I'll have a look at it this week.

**Comment by nacx** [comment]
@monitorjbl I've just opened https://github.com/monitorjbl/jclouds/pull/1 to your branch with an approach that works. HTH!

**Comment by monitorjbl** [comment]
Thanks for doing that! I thought you wanted me to test literally just that adapter, so I wouldn't have pulled in `NovaParserModule` without your pointer. I'll get a couple more test cases in there now.

How would you like me to handle the commit history? I'd guess you don't want that merge into my fork in there, so is it alright if I squash it down to one?

**Comment by nacx** [comment]
Sure! Forgey about my commit. It is only a proposal :) Feel free to squash and fix it as needed.

File: apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/Image.java [hunk 1]
```diff
@@ -85,6 +86,7 @@ public static Status fromValue(String v) {
       protected int minDisk;
       protected int minRam;
       protected Resource server;
+      protected List<BlockDeviceMapping> blockDeviceMapping;
```
Comment (everett-toews) [inline apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/Image.java]: This should be initialized with `ImmutableList.of()` (unless there's a really good reason not to). All of the other Collection types in the domain package are initialized this way. `blockDeviceMapping` should be the same.

File: apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/Image.java [hunk 2]
```diff
@@ -221,6 +229,7 @@ protected Image(String id, @Nullable String name, java.util.Set<Link> links, @Nu
       this.progress = progress;
       this.minDisk = minDisk;
       this.minRam = minRam;
+      this.blockDeviceMapping = blockDeviceMapping;
```
Comment (everett-toews) [inline apis/openstack-nova/src/main/java/org/jclouds/openstack/nova/v2_0/domain/Image.java]: Init the same as `metadata`

**Comment by monitorjbl** [comment]
Updated init for blockDeviceMapping.

**Comment by everett-toews** [comment]
[Checkstyle violation](https://jclouds.ci.cloudbees.com/job/jclouds-pull-requests/1527/org.apache.jclouds.api$openstack-nova/violations/file/src/test/java/org/jclouds/openstack/nova/v2_0/config/ImageAdapterTest.java/)

Unused import in `ImageAdapterTest.java`

**Comment by monitorjbl** [comment]
Whoops! Fixed it, sorry about that.

**Comment by everett-toews** [comment]
Merged to [1.8.x](http://git-wip-us.apache.org/repos/asf/jclouds/commit/f708d203) and [master](http://git-wip-us.apache.org/repos/asf/jclouds/commit/73600c81)

Nice work @monitorjbl! This was a tough one but you stuck with it. Can we give you a shout out on Twitter? If so, what's your Twitter handle?

Thanks for all the help @nacx

**Comment by monitorjbl** [comment]
Thanks, and I definitely don't mind a tweet :) My handle is @monitorjbl.

**Pull request closed without merging**